File: ConvertInchToMeterApp/MainActivity.java
package com.hayleydodkins.convertinchtometer;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    EditText input;
    Button calculateBtn;

    TextView resultText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        findViews();
        addClickListner();
    }

    private void findViews(){
        this.input = findViewById(R.id.inch_input);
        this.calculateBtn = findViewById(R.id.calculate_btn);
        this.resultText = findViewById(R.id.meter_text);
    }

    private void addClickListner(){
        //Add event listner to the button
        calculateBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                double meters = convertToMeters();
                displayResult(meters);
                Toast.makeText(MainActivity.this,"Calculating meters...",Toast.LENGTH_LONG).show();
            }
        });
    }

    private double convertToMeters(){
        double inches = Double.parseDouble(input.getText().toString());
        return inches*0.0254;
    }

    private void displayResult(double meters){
        resultText.setText(String.valueOf(meters));
    }
}

File: ShamelessSelfPromoApp/MainActivity.kt
package com.hayleydodkins.shamlessselfpromoapp

import android.content.Intent
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.CheckBox
import android.widget.Spinner
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.google.android.material.textfield.TextInputEditText




class MainActivity : AppCompatActivity() {

    private var contactNameInput : TextInputEditText? = null
    private var numberInput : TextInputEditText? = null
    private var displayNameInput : TextInputEditText? = null

    private var startDateInput : TextInputEditText? = null
    private var juniorCheckBox : CheckBox? = null
    private var immStartCheckBox : CheckBox? = null
    private var jobTitleInput : Spinner? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        val button : Button = findViewById(R.id.submit_button)

        button.setOnClickListener{
            Toast.makeText(this,"Creating Promo ...",Toast.LENGTH_LONG).show()
            onSubmitClick()
        }

        val spinnerValues : Array<String> = arrayOf("Software Engineer","Penetration Tester","Chief Information Security Officer","Network Engineer","Front-end Developer","Backend Developer","Game Designer")

        val spinnerAdapter = ArrayAdapter<String> (this,android.R.layout.simple_spinner_dropdown_item,spinnerValues)
        jobTitleInput  = findViewById(R.id.spinner_job_title)
        jobTitleInput?.adapter = spinnerAdapter
    }


    private fun onSubmitClick (){
        //Get the input data
         contactNameInput  = findViewById(R.id.edit_text_contact_name)
         numberInput  = findViewById(R.id.edit_text_contact_number)
         displayNameInput  = findViewById(R.id.edit_text_display_name)
        startDateInput  = findViewById(R.id.edit_text_start)
        juniorCheckBox  = findViewById(R.id.check_box_junior)
        immStartCheckBox  = findViewById(R.id.check_box_im_start)


        //Convert into strings
        val contactName = contactNameInput?.getText().toString()
        val number = numberInput?.getText().toString()
        val displayName = displayNameInput?.getText().toString()
        val date = startDateInput?.getText().toString()
        val junior = juniorCheckBox?.isChecked
        val immStart = immStartCheckBox?.isChecked
        val jobTitle = jobTitleInput?.selectedItem.toString()

        if(jobTitle != null && immStart != null && junior != null)
        {
            val textStr = "Contact Name: $contactName\n Contact Number: $number\nDisplay Name: $displayName\nStart Date: $date\nIs Junior: $junior\nImmidate Start: $immStart\nJob Title: $jobTitle"
        }

        //Create new explicit intent
        val previewActivityIntent = Intent(this,PreviewActivity::class.java)

        //Pass the data through
        previewActivityIntent.putExtra("contactName",contactName)
        previewActivityIntent.putExtra("contactNumber",number)
        previewActivityIntent.putExtra("displayName",displayName)
        previewActivityIntent.putExtra("startDate",date)
        previewActivityIntent.putExtra("junior",junior)
        previewActivityIntent.putExtra("immStart",immStart)
        previewActivityIntent.putExtra("jobTitle",jobTitle)

        //navigate to the new activity
        startActivity(previewActivityIntent)

    }
}